### ISSUE TITLE
Remove `_grab_charge`

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1731]
We have a function, `~plasmapy.formulary.misc._grab_charge` which grabs the charge of a particle. It will no longer be needed once #1057 is merged, and should be removed in favor of using `@particle_input`.  It is still used in a few places, so this will require some changes.

This was what I got from using `grep` for `grab_charge`.
```
formulary/braginskii.py:from plasmapy.formulary.misc import _grab_charge
formulary/braginskii.py:        self.Z = _grab_charge(ion, Z) * u.dimensionless_unscaled
formulary/frequencies.py:    Z = misc._grab_charge(particle, Z)
formulary/frequencies.py:        # TODO REPLACE WITH Z = np.abs(_grab_charge(particle, z_mean)), some bugs atm
formulary/misc.py:def _grab_charge(ion: Particle, z_mean=None):
formulary/parameters.py:    ("misc", "_grab_charge"),
formulary/speeds.py:    Z = misc._grab_charge(ion, z_mean)
formulary/tests/test_parameters.py:    _grab_charge,
formulary/tests/test_parameters.py:        ({"ion": "He+", "z_mean": 1.32}, _grab_charge, misc._grab_charge),
plasma/sources/plasmablob.py:from plasmapy.formulary.misc import _grab_charge
plasma/sources/plasmablob.py:        self.Z = _grab_charge(particle, Z)
```

I'm putting the status "on hold" for now, but this can be worked on once #1057 is done.
